File: Comparativa Python/Ejercicio7-Tablas.py
"""Desarrollar un programa que determine si una matriz es simétrica o no. Una
matriz es simétrica si es cuadrada y si es igual a su matriz transpuesta.

|8 1 3|     |8 1 3|
|1 7 4| ->  |1 7 4|
|3 4 9|     |3 4 9|
"""
import numpy as np

renglones = int(input('Digite el número de renglones: '))
columnas = int(input('Digite el número de columnas: '))

m = np.zeros((renglones, columnas), dtype=int)
# mT = np.zeros((renglones, columnas), dtype=int)

for r in range(renglones):
    for c in range(columnas):
        m[r][c] = int(input('Digite un número [{}][{}]: '.format(r+1, c+1)))

# ¿es matriz cuadrada?

if m.shape[0] == m.shape[1]:  # Es lo mismo que comparar renglones == columnas
    mT = m.transpose()  # matriz transpuesta
    if m.all() == mT.all():
        print('Es una matriz simétrica')
    else:
        print('No es una matriz simétrica')
else:
    print('No es una matriz simétrica')

File: Matrices o Tablas/Ejercicio7-Tablas.cpp
/*Desarrollar un programa que determine si una matriz es simétrica o no. Una matriz es simétrica si es cuadrada y si es igual a su matriz transpuesta.

|8 1 3|     |8 1 3|
|1 7 4| ->  |1 7 4|
|3 4 9|     |3 4 9|
*/

#include<iostream>
#include<stdlib.h>

using namespace std;

int main(){
    int numeros[100][100], filas, columnas;
    char band='F'; // Bandera

    cout<<"Digite el numero de filas: "; cin>>filas;
    cout<<"Digite el numero de columnas: "; cin>>columnas;

    for(int i=0;i<filas;i++){
        for(int j=0; j<columnas; j++){
            cout<<"Digite un numero ["<<i+1<<"]["<<j+1<<"]: ";
            cin>>numeros[i][j];
        }
    }
    if(filas==columnas){
        for(int i=0;i<filas;i++){
            for(int j=0;j<columnas;j++){
                if(numeros[i][j]==numeros[j][i]){
                    band='V';
                }
            }
        }
    }
    if (band == 'V'){
        cout<<"\nLa matriz es simetrica"<<endl;
    }
    else{
        cout<<"\nLa matriz no es simetrica"<<endl;
    }
 /* Mi versión   
    int matriz[100][100];
    int matrizt[100][100];
    int nFilas, nColumnas, contador = 0;

    cout<<"Digite numero de filas de la matriz: "; cin>>nFilas;
    cout<<"Digite numero de columnas de la matriz: "; cin>>nColumnas;

    for(int i=0;i<nFilas;i++){
        for(int j=0;j<nColumnas;j++){
            cout<<"Digite un numero ["<<i+1<<"]["<<j+1<<"]: ";
            cin>>matriz[i][j];
        }
    }

// verificando si es matriz cuadrada
    if (nFilas == nColumnas){
        // sacando matriz transpuesta
        for(int i=0; i<nFilas;i++){
            for(int j=0; j<nColumnas;j++){
                matrizt[i][j]=matriz[j][i];
            }
        }
        // Comparando matrices
        for(int i=0; i<nFilas;i++){
            for(int j=0; j<nColumnas;j++){
                if (matriz[i][j]==matrizt[i][j]){
                    contador++;
                };
            }
        }
        if (contador == nFilas*nColumnas){
            cout<<"\nLa matriz es simetrica"<<endl;
        }
        else{
            cout<<"\nLa matriz no es simétrica"<<endl;
        }
    }
    else {
        cout<<"\nLa matriz no es simétrica"<<endl;
    }*/
    system("pause");
    return 0;
}